modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -17,61 +17,125 @@
 
+    /// @private -- Exclude this function from OpenCV documentation
     GAPI_EXPORTS std::vector<std::string> getVectorOfStrings(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     template<typename... Types>
     cv::GCompileArgs getCompileArgs(const std::vector<char> &p);
 
+    /// @private -- Exclude this function from OpenCV documentation
     template<typename RMatAdapterType>
     cv::GRunArgs getRunArgsWithRMats(const std::vector<char> &p);
 } // namespace detail
 
+/** @brief This function allows to serialize GComputation.
+ *
+ * Check different overloads for more examples.
+ * @param c GComputation to serialize.
+ * @return serialized vector of bytes.
+ */
 GAPI_EXPORTS std::vector<char> serialize(const cv::GComputation &c);
-//namespace{
 
-template<typename T> static inline
-T deserialize(const std::vector<char> &p);
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const cv::GCompileArgs& ca);
 
-//} //ananymous namespace
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const cv::GMetaArgs& ma);
 
-GAPI_EXPORTS std::vector<char> serialize(const cv::GCompileArgs&);
-GAPI_EXPORTS std::vector<char> serialize(const cv::GMetaArgs&);
-GAPI_EXPORTS std::vector<char> serialize(const cv::GRunArgs&);
-GAPI_EXPORTS std::vector<char> serialize(const std::vector<std::string>&);
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const cv::GRunArgs& ra);
 
+//! @overload
+GAPI_EXPORTS std::vector<char> serialize(const std::vector<std::string>& vs);
+
+/**
+ * @private
+ */
+template<typename T> static inline
+T deserialize(const std::vector<char> &p);
+
+/** @brief This function allows to deserialize GComputation.
+ *
+ * Check different overloads for more examples.
+ * @param p serialized vector of bytes.
+ * @return deserialized GComputation object.
+ */
 template<> inline
 cv::GComputation deserialize(const std::vector<char> &p) {
     return detail::getGraph(p);
 }
 
+/** @brief This function allows to deserialize GMetaArgs.
+ *
+ * Check different overloads for more examples.
+ * @param p serialized vector of bytes.
+ * @return deserialized GMetaArgs object.
+ */
 template<> inline
 cv::GMetaArgs deserialize(const std::vector<char> &p) {
     return detail::getMetaArgs(p);
 }
 
+/** @brief This function allows to deserialize GRunArgs.
+ *
+ * Check different overloads for more examples.
+ * @param p serialized vector of bytes.
+ * @return deserialized GRunArgs object.
+ */
 template<> inline
 cv::GRunArgs deserialize(const std::vector<char> &p) {
     return detail::getRunArgs(p);
 }
 
+/**
+ * @private
+ */
 template<> inline
 std::vector<std::string> deserialize(const std::vector<char> &p) {
     return detail::getVectorOfStrings(p);
 }
 
+/**
+ * @brief This function is used to deserialize GCompileArgs which
+ * types were specified in the template.
+ *
+ * @note To be used properly all GCompileArgs types must be de serializable.
```

Comment:
What does it means? Ideally here you need to point to the structure one need to provide in order to make the type serializable.

Reply:
Rephrased